app/assets/javascripts/test/reducers/skeletontracing_reducer.spec.js

```diff
@@ -201,14 +214,23 @@ test("SkeletonTracing should delete several nodes from a tree", t => {
   );
   const deleteNodeAction = SkeletonTracingActions.deleteNodeAction();
 
-  // Add one node, then delete two times
-  const newState = SkeletonTracingReducer(initialState, createNodeAction);
-  const newStateA = SkeletonTracingReducer(newState, deleteNodeAction);
-  const newStateB = SkeletonTracingReducer(newStateA, deleteNodeAction);
+  // Create tree, add two nodes, then delete them again so that the tree is removed, as well
+  const emptyTreeState = SkeletonTracingReducer(
+    initialState,
+    SkeletonTracingActions.createTreeAction(),
+  );
 
-  t.not(newStateB, newState);
-  t.not(newStateA, newState);
-  t.is(newStateB, newStateA);
+  const newState = ChainReducer(emptyTreeState)
+    .apply(SkeletonTracingReducer, createNodeAction)
+    .apply(SkeletonTracingReducer, createNodeAction)
+    .apply(SkeletonTracingReducer, deleteNodeAction)
+    .apply(SkeletonTracingReducer, deleteNodeAction)
+    .unpack();
+
+  t.deepEqual(
```

Comment:
Shouldn't `t.deepEqual(emptyTreeState, newState)` be true here?

Reply:
Unfortunately not :( The DiffableMap node collection creates an internal chunk when nodes are added, which is not removed again (it's just emptied). That's why the states differ. Ideally, the equal method would not compare the internal state of the DiffableMap, but I don't know how we can achieve this consistently without doing magic in an own compare-function..

Reply:
I see, no big deal.